### PR TITLE
samples: bluetooth: Add support for nRF54L15DK

### PR DESCRIPTION
**Pull request by maje-emb**
Added nRF54L15DK support in the following samples:
  - central_bas
  - central_hr_coded
  - multiple_adv_sets
  - peripheral_bms
  - peripheral_cgms
  - peripheral_cts_client
  - peripheral_gatt_dm
  - peripheral_hr_coded
  - peripheral_mds
  - peripheral_nfc_pairing
  - peripheral_power_profiling
  - peripheral_rscs
  - shell_bt_nus
  - throughput

### CODE REVIEW

File: samples/bluetooth/peripheral_mds/README.rst [hunk 1]
```diff
@@ -33,54 +33,115 @@ The sample also includes the BAS functionalities.
...
+      * ``ncs_bt_rx_unused_stack`` - HCI RX thread stack.
+      * ``ncs_bt_tx_unused_stack`` - HCI TX thread stack.
 
 Error tracking with trace events
 ================================
 
-The sample implements the following user-defined trace reason for demonstration purposes:
+.. tabs::
+
+   .. group-tabs:: nRF52 and nRF53 DKs
```
Comment (peknis) [inline samples/bluetooth/peripheral_mds/README.rst:103]:
```suggestion
   .. group-tab:: nRF52 and nRF53 DKs
```

File: samples/bluetooth/peripheral_mds/README.rst [hunk 2]
```diff
@@ -33,54 +33,115 @@ The sample also includes the BAS functionalities.
...
+      * ``ncs_bt_connection_count`` - Number of Bluetooth connections.
+      * ``ncs_bt_connection_time_ms`` - Bluetooth connection time.
 
-  * ``ncs_bt_rx_unused_stack`` - HCI RX thread stack.
-  * ``ncs_bt_tx_unused_stack`` - HCI TX thread stack.
+         Time with at least one live Bluetooth connection.
+      * ``ncs_bt_bond_count`` - Number of Bluetooth bonds.
+
+      * Stack usage metrics shows the free stack space in bytes.
+      Configurable by the :kconfig:option:`CONFIG_MEMFAULT_NCS_STACK_METRICS` Kconfig option.
```
Comment (peknis) [inline samples/bluetooth/peripheral_mds/README.rst:63]:
```suggestion
      * ``ncs_bt_bond_count`` - Number of Bluetooth bonds.
      * Stack usage metrics shows the free stack space in bytes.
        Configurable by the :kconfig:option:`CONFIG_MEMFAULT_NCS_STACK_METRICS` Kconfig option.
```

File: samples/bluetooth/peripheral_mds/README.rst [hunk 3]
```diff
@@ -33,54 +33,115 @@ The sample also includes the BAS functionalities.
...
-The sample implements the following user-defined trace reason for demonstration purposes:
+.. tabs::
+
+   .. group-tabs:: nRF52 and nRF53 DKs
+
+      The sample implements the following user-defined trace reason for demonstration purposes:
+
+      ``button_state_changed`` - Collected every time when **Button 2** changes its state.
+
+      The trace events are defined in the file :file:`samples/bluetooth/peripheral_mds/memfault_config/memfault_trace_reason_user_config.def` file.
```
Comment (peknis) [inline samples/bluetooth/peripheral_mds/README.rst:109]:
```suggestion
      The trace events are defined in the :file:`samples/bluetooth/peripheral_mds/memfault_config/memfault_trace_reason_user_config.def` file.
```

File: samples/bluetooth/peripheral_mds/README.rst [hunk 4]
```diff
@@ -33,54 +33,115 @@ The sample also includes the BAS functionalities.
...
+      See `Memfault: Error Tracking with Trace Events`_ for more details about trace events.
+
+   .. group-tabs:: nRF54 DKs
```
Comment (peknis) [inline samples/bluetooth/peripheral_mds/README.rst:112]:
```suggestion
   .. group-tab:: nRF54 DKs
```

File: samples/bluetooth/peripheral_mds/README.rst [hunk 5]
```diff
@@ -33,54 +33,115 @@ The sample also includes the BAS functionalities.
...
+      The trace events are defined in the file :file:`samples/bluetooth/peripheral_mds/memfault_config/memfault_trace_reason_user_config.def` file.
+      See `Memfault: Error Tracking with Trace Events`_ for more details about trace events.
 
 Core dumps
 ==========
 
-You can trigger core dumps in this sample using the following methods:
+.. tabs::
+
+   .. group-tabs:: nRF52 and nRF53 DKs
```
Comment (peknis) [inline samples/bluetooth/peripheral_mds/README.rst:126]:
```suggestion
   .. group-tab:: nRF52 and nRF53 DKs
```

File: samples/bluetooth/peripheral_mds/README.rst [hunk 6]
```diff
@@ -33,54 +33,115 @@ The sample also includes the BAS functionalities.
...
+
+         * **Button 4** - Triggers a hardfault exception by division by zero.
+         * ``mfl crash`` shell command - Triggers an assertion fail.
+
+      When a fault occurs, it results in crashes that are captured by Memfault.
+      After your development kit reboots and reconnects with the Bluetooth gateway, it sends core dump data to the Memfault cloud for further inspection and analysis.
 
-   * **Button 4** - Triggers a hardfault exception by division by zero.
-   * ``mfl crash`` shell command - Triggers an assertion fail.
+   .. group-tabs:: nRF54 DKs
```
Comment (peknis) [inline samples/bluetooth/peripheral_mds/README.rst:136]:
```suggestion
   .. group-tab:: nRF54 DKs
```

File: samples/bluetooth/peripheral_mds/README.rst [hunk 7]
```diff
@@ -98,29 +159,57 @@ User interface
 The sample supports a simple user interface.
 You can control the sample using predefined buttons, while LEDs are used to display information.
 
-LED 1:
-   Blinks, toggling on/off every second, when the main loop is running and the device is advertising.
+.. tabs::
+
+   .. group-tabs:: nRF52 and nRF53 DKs
```
Comment (peknis) [inline samples/bluetooth/peripheral_mds/README.rst:164]:
```suggestion
   .. group-tab:: nRF52 and nRF53 DKs
```

File: samples/bluetooth/peripheral_mds/README.rst [hunk 8]
```diff
@@ -98,29 +159,57 @@ User interface
...
+
+         During the pairing procedure, press this button to reject pairing.
+
+      Button 3:
+         Every press of this button is counted under the ``button_press_count`` metric.
+
+      Button 4:
+         Simulate a development kit crash by triggering a hardfault exception by division by zero.
+
+   .. group-tabs:: nRF54 DKs
```
Comment (peknis) [inline samples/bluetooth/peripheral_mds/README.rst:189]:
```suggestion
   .. group-tab:: nRF54 DKs
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/bluetooth/peripheral_mds/README.rst [hunk 9]
```diff
@@ -33,54 +33,114 @@ The sample also includes the BAS functionalities.
...
+      * Bluetooth metrics, enabled and disabled using the :kconfig:option:`CONFIG_MEMFAULT_NCS_BT_METRICS` Kconfig option.
+
+      * ``ncs_bt_connection_count`` - Number of Bluetooth connections.
+      * ``ncs_bt_connection_time_ms`` - Bluetooth connection time.
+
+         Time with at least one live Bluetooth connection.
+      * ``ncs_bt_bond_count`` - Number of Bluetooth bonds.
+
+      * Stack usage metrics shows the free stack space in bytes.
+      Configurable by the :kconfig:option:`CONFIG_MEMFAULT_NCS_STACK_METRICS` Kconfig option.
```
Comment (peknis) [inline samples/bluetooth/peripheral_mds/README.rst:92]:
```suggestion
      * Stack usage metrics shows the free stack space in bytes.
        Configurable by the :kconfig:option:`CONFIG_MEMFAULT_NCS_STACK_METRICS` Kconfig option.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/bluetooth/peripheral_mds/README.rst [hunk 10]
```diff
@@ -33,54 +33,114 @@ The sample also includes the BAS functionalities.
...
 
-* ``button_3_press_count`` - The number of **Button 3** presses.
-* ``battery_soc_pct`` - The simulated battery level.
-* ``button_1_elapsed_time_ms`` - The time measured between two **Button 1** presses.
+   .. group-tab:: nRF52 and nRF53 DKs
 
-These metrics are defined in the :file:`samples/bluetooth/peripheral_mds/memfault_config/memfault_metrics_heartbeat_config.def` file.
-For more details about the metrics, see `Memfault: Collecting Device Metrics`_.
+      The sample shows how to capture user-specific metrics.
+      It defines the following metrics:
```
Comment (peknis) [inline samples/bluetooth/peripheral_mds/README.rst:41]: Move this introductory text outside the tab, so you do not need to repeat it for both group-tabs. And do the same in other two sections below. Like this:

The sample shows how to capture...

.. tabs::

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Review by peknis — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

File: samples/bluetooth/peripheral_nfc_pairing/README.rst [hunk 11]
```diff
@@ -136,23 +136,13 @@ Testing with NFC Poller Device
...
-      1. Touch the NFC antenna with the NFC Poller Device, for example :ref:`central_nfc_pairing` and observe that **LED 1** is lit.
-      #. Observe the output log.
-         The content of the exchanged NDEF messages is printed there.
-      #. Check the security level of the paring on the terminal.
-      #. Move the NFC antenna away from the NFC Poller Device.
+   1. Touch the NFC antenna with the NFC Poller Device, for example :ref:`central_nfc_pairing` and observe that **LED 2** is lit.
+   #. Observe the output log.
+      The content of the exchanged NDEF messages is printed there.
+   #. Check the security level of the paring on the terminal.
+   #. Move the NFC antenna away from the NFC Poller Device.
```
Comment (dchat-nordic) [inline samples/bluetooth/peripheral_nfc_pairing/README.rst:145]: Are the steps for testing with nRF54 missing, or were they moved?

**Review by dchat-nordic — CHANGES_REQUESTED** [review]

File: samples/bluetooth/peripheral_nfc_pairing/README.rst (hunk 11 above)
Comment (maje-emb) [reply samples/bluetooth/peripheral_nfc_pairing/README.rst:145]: `central_nfc_pairing` is not supported on nRF54, therefore these test steps have been removed for nRF54.

**Review by dchat-nordic — APPROVED** [review]

File: samples/bluetooth/throughput/src/main.c [hunk 12]
```diff
@@ -650,8 +650,14 @@ int main(void)
...
-	printk("Press button 1 or type \"central\" on the central board.\n");
-	printk("Press button 2 or type \"peripheral\" on the peripheral board.\n");
+
+	if (IS_ENABLED(CONFIG_SOC_SERIES_NRF54HX) || IS_ENABLED(CONFIG_SOC_SERIES_NRF54LX)) {
+		printk("Press button 0 or type \"central\" on the central board.\n");
+		printk("Press button 1 or type \"peripheral\" on the peripheral board.\n");
+	} else {
+		printk("Press button 1 or type \"central\" on the central board.\n");
+		printk("Press button 2 or type \"peripheral\" on the peripheral board.\n");
+	}
```
Comment (doki-nordic) [inline samples/bluetooth/throughput/src/main.c:660]: Maybe some future improvement: Create some common defines that contains names for each led and button on each board.

**Review by doki-nordic — CHANGES_REQUESTED** [review]

File: samples/bluetooth/peripheral_mds/prj.conf [hunk 13]
```diff
@@ -38,8 +38,8 @@ CONFIG_HEAP_MEM_POOL_SIZE=256
 CONFIG_MEMFAULT=y
 CONFIG_MEMFAULT_SHELL=y
 
-CONFIG_MEMFAULT_NCS_PROJECT_KEY=""
-CONFIG_MEMFAULT_NCS_DEVICE_ID=""
+CONFIG_MEMFAULT_NCS_PROJECT_KEY="asd"
+CONFIG_MEMFAULT_NCS_DEVICE_ID="asdf"
```
[the remark this answers is not in the document]
Comment (maje-emb) [reply samples/bluetooth/peripheral_mds/prj.conf:42]: An overlook, this remained from the tests.

**Review by doki-nordic — APPROVED** [review]

**Comment by rlubos** [comment]
@nrfconnect/ncs-dragoon Ping

**Review by ppryga-nordic — APPROVED** [review]
Approved. FYI i didn't review the `samples/bluetooth/peripheral_mds/README.rst`

**Pull request merged**